src/sdk/namespace_fs.js

```diff
@@ -2773,7 +2773,11 @@ class NamespaceFS {
                     await native_fs_utils.safe_unlink(fs_context, file_path, version_info, gpfs_options, bucket_tmp_dir_path);
                     return { ...version_info, latest: true };
                 } else {
-                    await nb_native().fs.unlink(fs_context, file_path);
+                    try {
+                        await nb_native().fs.unlink(fs_context, file_path);
+                    } catch (err) {
+                        if (err.code !== 'ENOENT') throw err;
+                    }
```

Comment:
Could you add a helper function `unlink_deleted_object` (for example)? (hoping that we will not miss more cases with it)